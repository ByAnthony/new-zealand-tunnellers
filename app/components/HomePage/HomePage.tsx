"use client";

import { HistoryChapters } from "./HistoryChapters/HistoryChapters";
import { TunnellersImages } from "./TunnellersImages/TunnellersImages";

import STYLES from "./HomePage.module.scss";

type Props = {
  homepage: any;
};

export function HomePage({ homepage }: Props) {
  return (
    <div className={STYLES["homepage-container"]}>
      <div className={STYLES.intro}>
        <h1>
          The Kiwis who
          <br />
          fought underground
          <br />
          during World War I
        </h1>
      </div>
      <TunnellersImages images={homepage.tunnellers} />
      <HistoryChapters articles={homepage.historyChapters} />
    </div>
  );
}
