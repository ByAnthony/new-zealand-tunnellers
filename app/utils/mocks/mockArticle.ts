import { AboutUsArticle } from "../../types/article";

export const mockArticle: AboutUsArticle = {
  id: "about-us",
  title: "Our Story",
  section: [
    { title: "Title 1", text: "This is the first paragraph of our story." },
    { title: "Title 2", text: "This is the second paragraph of our story." },
  ],
  image: [
    {
      file: "our-story.jpg",
      title: null,
      photographer: null,
      reference: null,
      alt: "A picture of our team",
    },
  ],
};
