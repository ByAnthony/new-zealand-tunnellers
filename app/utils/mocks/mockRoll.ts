import { Tunneller } from '@/app/types/roll';
import { mockId } from './mockProfile';

const johnDoe = {
  forename: 'John',
  surname: 'Doe',
};

const alexanderDriscott = {
  forename: 'Alexander',
  surname: 'Driscott',
};

const williamRight = {
  forename: 'William',
  surname: 'Right',
};

const robertLang = {
  forename: 'Robert',
  surname: 'Lang',
};

export const mockDetails: Tunneller = {
  id: mockId,
  name: johnDoe,
  birthDate: '1886',
  deathDate: '1952',
};

export const mockDetailsList: Tunneller[] = [
  mockDetails,
  {
    ...mockDetails,
    id: 1,
    name: alexanderDriscott,
  },
];

export const mockRoll: Record<string, Tunneller[]> = {
  D: [
    mockDetails,
    {
      ...mockDetails,
      id: 1,
      name: alexanderDriscott,
    },
  ],
  L: [
    {
      ...mockDetails,
      id: 2,
      name: robertLang,
    },
  ],
  R: [
    {
      ...mockDetails,
      id: 3,
      name: williamRight,
    },
  ],
};
