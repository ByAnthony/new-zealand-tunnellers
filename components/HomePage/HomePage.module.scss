@use "app/variables.module.scss";

.homepage-container {
  margin: 0 auto;

  @include variables.desktop {
    width: variables.$inner-width;
  }
}
