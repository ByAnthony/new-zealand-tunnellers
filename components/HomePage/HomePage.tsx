"use client";

import { History } from "@/components/HomePage/History/History";
import { Tunnellers } from "@/components/HomePage/Tunnellers/Tunnellers";

import STYLES from "./HomePage.module.scss";

type Props = {
  homepage: any;
};

export function HomePage({ homepage }: Props) {
  return (
    <div className={STYLES["homepage-container"]}>
      <Tunnellers />
      <History articles={homepage.historyChapters} />
    </div>
  );
}
