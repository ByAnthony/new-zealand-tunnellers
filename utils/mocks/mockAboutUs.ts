import { AboutUsArticle } from "@/types/article";

import {
  mockSectionOne,
  mockSectionTwo,
  mockTitle,
  mockTopImage,
} from "./mockArticle";

export const mockAboutUs: AboutUsArticle = {
  id: "my-awesome-article",
  title: mockTitle,
  section: [mockSectionOne, mockSectionTwo],
  image: [mockTopImage],
};
