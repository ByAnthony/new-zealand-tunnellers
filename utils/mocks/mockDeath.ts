import { Death } from "@/types/tunneller";

import { mockDeath } from "./mockTunneller";

export const mockAfterWarDeath: Death = {
  ...mockDeath,
  warInjuriesDeathAfterWar: true,
};
