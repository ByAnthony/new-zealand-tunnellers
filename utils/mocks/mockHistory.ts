import { HistoryChapter } from "@/types/homepage";

export const mockHistory: HistoryChapter[] = [
  {
    id: "where-all-started",
    chapter: 1,
    title: "Where All\\Started",
    image: "where-all-started.jpg",
  },
  {
    id: "the-journey-ahead",
    chapter: 2,
    title: "The Journey\\Ahead",
    image: "the-journey-ahead.jpg",
  },
  {
    id: "discovering-the-war",
    chapter: 3,
    title: "Discovering\\The War",
    image: "discovering-the-war.jpg",
  },
  {
    id: "below-the-surface",
    chapter: 4,
    title: "Below\\The Surface",
    image: "below-the-surface.jpg",
  },
  {
    id: "back-to-the-surface",
    chapter: 5,
    title: "Back\\To The Surface",
    image: "back-to-the-surface.jpg",
  },
  {
    id: "reconversion-of-activities",
    chapter: 6,
    title: "Reconversion\\Of Activities",
    image: "reconversion-of-activities.jpg",
  },
  {
    id: "returnin-home",
    chapter: 7,
    title: "Returning\\Home",
    image: "returning-home.jpg",
  },
];
