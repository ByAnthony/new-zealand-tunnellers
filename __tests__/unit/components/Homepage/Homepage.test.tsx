import { render } from "@testing-library/react";

import { HomePage } from "@/components/HomePage/HomePage";
import { mockHistory } from "@/utils/mocks/mockHistory";

describe("Homepage", () => {
  const mockHomepage = {
    historyChapters: mockHistory,
  };

  test("matches the snapshot", () => {
    const { asFragment } = render(<HomePage homepage={mockHomepage} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
