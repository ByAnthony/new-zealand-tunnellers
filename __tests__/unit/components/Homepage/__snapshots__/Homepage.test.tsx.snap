// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage matches the snapshot for mobile viewport 1`] = `
<DocumentFragment>
  <div
    class="homepage-container"
  >
    <div
      class="intro"
    >
      <svg
        viewBox="0 0 100 125"
      >
        <defs>
          <mask
            height="125"
            id="mask"
            width="100"
            x="0"
            y="0"
          >
            <rect
              fill="white"
              height="125"
              width="100"
              x="0"
              y="0"
            />
            <text
              text-anchor="start"
              x="0"
              y="10"
            >
              <tspan
                dy="0.8em"
                x="0"
              >
                The Kiwis
              </tspan>
              <tspan
                dy="0.8em"
                x="0"
              >
                who
              </tspan>
              <tspan
                dy="0.8em"
                x="0"
              >
                fought
              </tspan>
              <tspan
                dy="0.8em"
                x="0"
              >
                beneath
              </tspan>
              <tspan
                dy="0.8em"
                x="0"
              >
                the no
              </tspan>
              <tspan
                dy="0.8em"
                x="0"
              >
                man’s
              </tspan>
              <tspan
                dy="0.8em"
                x="0"
              >
                land
              </tspan>
            </text>
          </mask>
        </defs>
        <rect
          fill="rgb(24, 26, 27)"
          height="125"
          mask="url(#mask)"
          width="100"
          x="0"
          y="0"
        />
      </svg>
    </div>
    <div
      class="roll-wrapper"
    >
      <div
        class="roll-container"
      >
        <a
          class="roll"
          href="/tunnellers"
        >
          <span
            class="discover"
          >
            Discover
          </span>
          The New Zealand Tunnellers
        </a>
      </div>
    </div>
    <div
      class="history-chapter"
    >
      <div
        class="chapter-cards-wrapper"
        id="history"
      >
        <div
          class="chapter-cards-menu"
        >
          <h2>
            History of the Company
          </h2>
          <div
            class="chapter-cards-nav"
          >
            <button
              aria-label="Scroll to the left to see previous chapters"
              disabled=""
              style="color: rgb(44, 46, 47); border-color: rgb(44, 46, 47); background-color: rgb(24, 26, 27); cursor: not-allowed;"
              type="button"
            >
              ←
            </button>
            <button
              aria-label="Scroll to the right to see next chapters"
              type="button"
            >
              →
            </button>
          </div>
        </div>
        <div
          class="chapter-cards"
          data-testid="chapters-scroll"
        >
          <div
            class="chapter-card"
            style="background-image: url(/images/history/where-all-started.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 1: Where All Started"
              class="link-button"
              href="/history/where-all-started"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Where All
                    </span>
                    <span
                      class="title-line-2"
                    >
                      Started
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 1
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/the-journey-ahead.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 2: The Journey Ahead"
              class="link-button"
              href="/history/the-journey-ahead"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      The Journey
                    </span>
                    <span
                      class="title-line-2"
                    >
                      Ahead
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 2
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/discovering-the-war.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 3: Discovering The War"
              class="link-button"
              href="/history/discovering-the-war"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Discovering
                    </span>
                    <span
                      class="title-line-2"
                    >
                      The War
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 3
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/below-the-surface.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 4: Below The Surface"
              class="link-button"
              href="/history/below-the-surface"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Below
                    </span>
                    <span
                      class="title-line-2"
                    >
                      The Surface
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 4
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/back-to-the-surface.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 5: Back To The Surface"
              class="link-button"
              href="/history/back-to-the-surface"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Back
                    </span>
                    <span
                      class="title-line-2"
                    >
                      To The Surface
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 5
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/reconversion-of-activities.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 6: Reconversion Of Activities"
              class="link-button"
              href="/history/reconversion-of-activities"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Reconversion
                    </span>
                    <span
                      class="title-line-2"
                    >
                      Of Activities
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 6
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/returning-home.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 7: Returning Home"
              class="link-button"
              href="/history/returnin-home"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Returning
                    </span>
                    <span
                      class="title-line-2"
                    >
                      Home
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 7
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Homepage matches the snapshot for tablet/desktop viewports 1`] = `
<DocumentFragment>
  <div
    class="homepage-container"
  >
    <div
      class="intro"
    >
      <svg
        viewBox="0 0 100 60"
      >
        <defs>
          <mask
            height="60"
            id="mask"
            width="100"
            x="0"
            y="0"
          >
            <rect
              fill="white"
              height="60"
              width="100"
              x="0"
              y="0"
            />
            <text
              text-anchor="start"
              x="0"
              y="10"
            >
              <tspan
                dy="0.8em"
                x="0"
              >
                The Kiwis
              </tspan>
              <tspan
                dy="0.8em"
                x="0"
              >
                who fought
              </tspan>
              <tspan
                dy="0.8em"
                x="0"
              >
                beneath the
              </tspan>
              <tspan
                dy="0.8em"
                x="0"
              >
                no man’s land
              </tspan>
            </text>
          </mask>
        </defs>
        <rect
          fill="rgb(24, 26, 27)"
          height="60"
          mask="url(#mask)"
          width="100"
          x="0"
          y="0"
        />
      </svg>
    </div>
    <div
      class="roll-wrapper"
    >
      <div
        class="roll-container"
      >
        <a
          class="roll"
          href="/tunnellers"
        >
          <span
            class="discover"
          >
            Discover
          </span>
          The New Zealand Tunnellers
        </a>
      </div>
    </div>
    <div
      class="history-chapter"
    >
      <div
        class="chapter-cards-wrapper"
        id="history"
      >
        <div
          class="chapter-cards-menu"
        >
          <h2>
            History of the Company
          </h2>
          <div
            class="chapter-cards-nav"
          >
            <button
              aria-label="Scroll to the left to see previous chapters"
              disabled=""
              style="color: rgb(44, 46, 47); border-color: rgb(44, 46, 47); background-color: rgb(24, 26, 27); cursor: not-allowed;"
              type="button"
            >
              ←
            </button>
            <button
              aria-label="Scroll to the right to see next chapters"
              type="button"
            >
              →
            </button>
          </div>
        </div>
        <div
          class="chapter-cards"
          data-testid="chapters-scroll"
        >
          <div
            class="chapter-card"
            style="background-image: url(/images/history/where-all-started.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 1: Where All Started"
              class="link-button"
              href="/history/where-all-started"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Where All
                    </span>
                    <span
                      class="title-line-2"
                    >
                      Started
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 1
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/the-journey-ahead.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 2: The Journey Ahead"
              class="link-button"
              href="/history/the-journey-ahead"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      The Journey
                    </span>
                    <span
                      class="title-line-2"
                    >
                      Ahead
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 2
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/discovering-the-war.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 3: Discovering The War"
              class="link-button"
              href="/history/discovering-the-war"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Discovering
                    </span>
                    <span
                      class="title-line-2"
                    >
                      The War
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 3
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/below-the-surface.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 4: Below The Surface"
              class="link-button"
              href="/history/below-the-surface"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Below
                    </span>
                    <span
                      class="title-line-2"
                    >
                      The Surface
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 4
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/back-to-the-surface.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 5: Back To The Surface"
              class="link-button"
              href="/history/back-to-the-surface"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Back
                    </span>
                    <span
                      class="title-line-2"
                    >
                      To The Surface
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 5
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/reconversion-of-activities.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 6: Reconversion Of Activities"
              class="link-button"
              href="/history/reconversion-of-activities"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Reconversion
                    </span>
                    <span
                      class="title-line-2"
                    >
                      Of Activities
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 6
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
          <div
            class="chapter-card"
            style="background-image: url(/images/history/returning-home.jpg); background-size: cover; background-position: center center;"
          >
            <a
              aria-label="Go to Chapter 7: Returning Home"
              class="link-button"
              href="/history/returnin-home"
            >
              <div
                class="chapter-card-dimmer"
              >
                <div
                  class="chapter-card-content"
                >
                  <div>
                    <span
                      class="title-line-1"
                    >
                      Returning
                    </span>
                    <span
                      class="title-line-2"
                    >
                      Home
                    </span>
                    <span
                      class="title-line-3"
                    >
                      Chapter 7
                    </span>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
